src/pydata_sphinx_theme/assets/styles/components/_navbar-links.scss

```diff
@@ -1,16 +1,10 @@
 /**
  * Navigation links in the navbar and icon links
  */
-.navbar-nav,
-.navbar-icon-links {
```

Comment:
I added the `.navbar-icon-links` selector in #1846 but I really wanted to completely separate `navbar-icon-links` from `navbar-nav`. I was able to finish that work in this PR.